modules/post/windows/gather/enum_domain_tokens.rb

```diff
@@ -37,188 +37,179 @@ def initialize(info = {})
+      'Users'
+    ].each do |group|
+      group_users = get_members_from_localgroup(group)
 
-    user_impersonation.each do |dt|
-      next unless dt =~ /#{domain}/
+      next unless group_users
 
-      user = dt.split("\\")[1]
-      if dom_admins.include?(user)
-        tbl << ["Impersonation", "User", dt, true]
-      else
-        tbl << ["Impersonation", "User", dt, false]
-      end
-    end
+      vprint_status("Group '#{group}' members: #{group_users.join(', ')}")
 
-    group_delegation.each do |dt|
-      next unless dt =~ /#{domain}/
+      group_users.each do |group_user|
+        next unless group_user.include?(domain)
 
-      user = dt.split("\\")[1]
-      if dom_admins.include?(user)
-        tbl << ["Delegation", "Group", dt, true]
-      else
-        tbl << ["Delegation", "Group", dt, false]
+        user = group_user.split('\\')[1]
+        tbl << [group, group_user, @domain_admins.include?(user)]
       end
     end
 
-    group_impersonation.each do |dt|
-      next unless dt =~ /#{domain}/
-
-      user = dt.split("\\")[1]
-      if dom_admins.include?(user)
-        tbl << ["Impersonation", "Group", dt, true]
-      else
-        tbl << ["Impersonation", "Group", dt, false]
-      end
-    end
-    results = tbl.to_s
-    print_line("\n" + results + "\n")
+    print_line("\n#{tbl}\n")
   end
 
-  def list_group_members(domain, dom_admins)
+  def list_tokens(domain)
     tbl = Rex::Text::Table.new(
-      'Header' => "Account in Local Groups with Domain Context",
+      'Header' => 'Impersonation Tokens with Domain Context',
       'Indent' => 1,
       'Columns' =>
       [
-        "Group",
-        "Member",
-        "Domain Admin"
+        'Token Type',
+        'Account Type',
+        'Account Name',
+        'Domain Admin'
       ]
     )
-    print_status("Checking local groups for Domain Accounts and Groups")
-    admins = get_members_from_localgroup("Administrators")
-    users = get_members_from_localgroup("users")
-    backops = get_members_from_localgroup("\"Backup Operators\"")
-    admins.each do |dt|
-      next unless dt =~ /#{domain}/
-
-      user = dt.split("\\")[1]
-      if dom_admins.include?(user)
-        tbl << ["Administrators", dt, true]
-      else
-        tbl << ["Administrators", dt, false]
-      end
+    print_status('Checking for Domain group and user tokens')
+
+    client.core.use('incognito') unless client.ext.aliases.include?('incognito')
+
+    user_tokens = client.incognito.incognito_list_tokens(0)
+    user_delegation = user_tokens['delegation'].split("\n")
+    user_impersonation = user_tokens['impersonation'].split("\n")
+
+    user_delegation.each do |dt|
+      next unless dt.include?(domain)
+
+      user = dt.split('\\')[1]
+      tbl << ['Delegation', 'User', dt, @domain_admins.include?(user)]
     end
 
-    backops.each do |dt|
-      next unless dt =~ /#{domain}/
+    user_impersonation.each do |dt|
+      next if dt == 'No tokens available'
```

Comment:
Just to confirm, is this string coming from the server? Do we get back different text if the server is in french or something else non-english 👀

Reply:
> Just to confirm, is this string coming from the server? Do we get back different text if the server is in french or something else non-english eyes

This is Meterperet code, so no.

```
c/meterpreter/source/extensions/python/Lib/meterpreter/incognito.py:7:INCOGNITO_NO_TOKENS = 'No tokens available\n'
```